src/main/scala/name/nikiforo/Simulate.scala

```diff
@@ -0,0 +1,43 @@
+package name.nikiforo
+
+import java.net.InetSocketAddress
+
+import cats.effect.ExitCode
+import cats.effect.IO
+import cats.effect.IOApp
+import cats.syntax.applicativeError._
+import cats.syntax.option._
+import fs2.Stream
+import fs2.concurrent.Broadcast
+
+object Simulate extends IOApp {
+
+  private val H42: Byte = 42
+
+  private val H24: Byte = 24
+
+  def run(args : List[String]): IO[ExitCode] =
+    Stream(Stream(H42, H24))
+      .repeat
+      .flatMap(_.through(clientPipe).handleErrorWith(logError))
+      .evalMap(_ => IO.unit)
```

Comment:
Surprisingly, this line is required for the error to happen.
This might be why without `socket.write` error wasn't reproducible.

Reply:
I just noticed the same! So strange. I've further minimized a bit:

```scala
    Stream(Stream(0, 1))
      .repeat
      .flatMap { s =>
        s.flatMap {
          case 0 => Stream(())
          case 1 => Stream.raiseError[IO](new RuntimeException)
        }.through(Broadcast(1)).pull.take(1).void.streamNoScope.foldMap(List(_)).flatMap(_.head)
      }
      .evalMap(_ => IO.unit)
      .compile
      .drain
      .recoverWith { ex => IO.delay(println(s"DISASTER: ${ex.getMessage}")) }
      .as(ExitCode.Success)
```

Reply:
@mpilquist, hm... There is no recover in your case

Reply:
```
    Stream(Stream(0, 1))
      .repeat
      .flatMap { s =>
        s.flatMap {
          case 0 => Stream(())
          case 1 => Stream.raiseError[IO](new RuntimeException)
        }
        .through(Broadcast(1))
        .pull
        .take(1)
        .void
        .streamNoScope
        .foldMap(List(_))
        .flatMap(_.head)
        .recoverWith { ex => Stream.eval_(IO.delay(println(s"catched: ${ex.getMessage}")))}
      }
      .evalMap(_ => IO.unit)
      .compile
      .drain
      .recoverWith { ex => IO.delay(println(s"DISASTER: ${ex.getMessage}")) }
      .as(ExitCode.Success)
```

Reply:
Ha yeah, accidentally lost it when moving stuff around. I just updated with a new version that still reproduces issue but is a bit simpler.

Reply:
Further simplified, without `Broadcast` or `pull.take(1)`:

```scala
    Stream(Stream(()) ++ Stream.raiseError[IO](new RuntimeException))
      .repeat
      .flatMap { s =>
        s.prefetch.handleErrorWith(t => Stream.empty)
      }
      .flatMap(_ => Stream.empty)
      .compile
      .drain
      .recoverWith { ex => IO.delay(println(s"DISASTER: ${ex.getMessage}")) }
      .as(ExitCode.Success)
```